### ISSUE TITLE
Template(named:inBundle:) crash

### ISSUE THREAD

**Issue by AliSoftware** [kylef/Stencil#31]
When trying to call:
```swift
let testBundle = NSBundle(forClass: StencilTests.self)
let template = try Template(named: "images.genum", inBundle: testBundle)
```
The code crashes with an `EXC_BAD_ACCESS`. The stacktrace suggests its a memory issue, with memory being freed while already deallocated.

Changing the code to this doesn't trigger the issue anymore:
```swift
let testBundle = NSBundle(forClass: StencilTests.self)
let path = Path(testBundle.pathForResource("images", ofType: "genum")!)
let template = try Template(path: path)
```

I see that `Template(named:inBundle:)` actually calls `URLForResource:withExtension:` — creating a `file:///` URL — and then use its `absoluteString` to create a `Path` and call the `Template(path:)` initializer.

It seems that using a fileURL's `absolutePath` like `file:///` as a `Path` doesn't please either `PathKit` or `Stencil` and makes it crash (while throwing and thus `deinit`?) when trying to read from it.
<img width="500" alt="stencil-stacktrace" src="https://cloud.githubusercontent.com/assets/216089/10562087/663642da-7549-11e5-8a69-9977f2bda3eb.png">

**Comment by kylef** [comment]
In 4d76fb4 I've added tests to ensure it works as expected, it does appear to work as expected with no changes when the file name is found in the given bundle.

When the file is not found, this would cause an error as you've experienced which I've fixed with f0abd34c320751cdb82b25e66db9b6a782fb27f6.

@AliSoftware can you confirm this has solved your problem? I don't see any issues related to `file:///` and it was never using `absoluteString` on the `NSURL` but `path` as expected. I think perhaps this issue got confused with not handling failures correctly and came to the wrong conclusion?

**Comment by AliSoftware** [comment]
> I don't see any issues related to file:/// and it was never using absoluteString on the NSURL but path as expected

I just understood why I got the problem. This *was* indeed due to the use of `absoluteString` instead of `path` on `file:///` URLs.

The reason why you didn't manage to reproduce is because… it has actually already been fixed by #29!

As my `Podfile` didn't point to any specific commit, I was still using `0.3.0` which was still using `absoluteString`. Pointing to `:git => …, :branch => 'master'` made the problem disappear.
